### PR TITLE
[data][api] mark ActorPoolStrategy as an internal API

### PR DESCRIPTION
**Pull request by can-anyscale**
`ActorPoolStrategy` is inside the `_internal` package, it shouldn't be a public api

Test:
- CI

### CODE REVIEW

**Review by bveeramani — APPROVED** [review]

**Comment by aslonnie** [comment]
cc @ray-project/ray-ci

**Comment by aslonnie** [comment]
I think it is good if data team thinks it is good.

File: python/ray/data/_internal/compute.py [hunk 1]
```diff
@@ -50,7 +50,6 @@ def __eq__(self, other: Any) -> bool:
         )
 
 
-@PublicAPI
```
Comment (aslonnie) [inline python/ray/data/_internal/compute.py:53]: ooc, I thought we say every API symbol needs to get an annotation?

Comment (can-anyscale) [reply python/ray/data/_internal/compute.py:53]: except for everything inside the `_internal` package and those with with `_` in its name

**Pull request merged**